parity-clib/examples/cpp/CMakeLists.txt

```diff
@@ -1,18 +1,25 @@
 cmake_minimum_required(VERSION 3.5)
 include(ExternalProject)
 include_directories("${CMAKE_SOURCE_DIR}/../..")
-set (CMAKE_CXX_STANDARD 11) # Enforce C++11
+set (CMAKE_CXX_STANDARD 17) # Enforce C++17
+set (CMAKE_CXX_STANDARD_REQUIRED ON) # Enforce C++17
+set (CMAKE_CXX_EXTENSIONS OFF) # Enforce C++17
+set (CMAKE_EXPORT_COMPILE_COMMANDS ON) # generate a compilation database
 add_executable(parity-example main.cpp)
-
+add_compile_options("-Wall -Wextra -Werror")
+set(parity_shared_library_path "${CMAKE_SOURCE_DIR}/../../../target/release/${CMAKE_SHARED_LIBRARY_PREFIX}parity${CMAKE_SHARED_LIBRARY_SUFFIX}")
 ExternalProject_Add(
     libparity
     DOWNLOAD_COMMAND ""
     CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    COMMAND cargo build -p parity-clib --verbose --color=always    # Note: use --release in a real project
+    BUILD_COMMAND cargo build -p parity-clib --release --verbose --color=always
```

Comment:
I'm not sure we want `--release` here because it is part of the `CI` pipeline and might significantly slower